File: app/src/main/java/ru/melodin/fast/api/model/VKChat.java
package ru.melodin.fast.api.model;

import org.json.JSONArray;
import org.json.JSONObject;

import java.io.Serializable;
import java.util.ArrayList;

import ru.melodin.fast.util.ArrayUtil;

public class VKChat extends VKModel implements Serializable {
    private static final long serialVersionUID = 1L;

    private int id;
    private String title;
    private int adminId;
    private ArrayList<VKUser> users = new ArrayList<>();
    private String photo50, photo100, photo200;

    private VKConversation.State state = VKConversation.State.IN;
    private VKConversation.Type type = VKConversation.Type.USER;

    public VKChat() {
    }

    public VKChat(JSONObject o) {
        this.id = o.optInt("id", -1);
        this.title = o.optString("title");
        this.adminId = o.optInt("admin_id", -1);

        JSONArray users = o.optJSONArray("users");
        if (!ArrayUtil.INSTANCE.isEmpty(users))
            this.users = VKUser.parse(users);

        this.photo50 = o.optString("photo_50");
        this.photo100 = o.optString("photo_100");
        this.photo200 = o.optString("photo_200");
        this.state = o.has("left") ? VKConversation.State.LEFT : o.has("kicked") ? VKConversation.State.KICKED : VKConversation.State.IN;
        this.type = VKConversation.getType(o.optString("type"));
    }

    public static int getIntState(VKConversation.State state) {
        if (state == null) return -1;
        switch (state) {
            case IN:
                return 0;
            case LEFT:
                return 1;
            case KICKED:
                return 2;
            default:
                return -1;
        }
    }

    public static VKConversation.State getState(int state) {
        switch (state) {
            default:
            case 0:
                return VKConversation.State.IN;
            case 1:
                return VKConversation.State.LEFT;
            case 2:
                return VKConversation.State.KICKED;
        }
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getAdminId() {
        return adminId;
    }

    public void setAdminId(int adminId) {
        this.adminId = adminId;
    }

    public ArrayList<VKUser> getUsers() {
        return users;
    }

    public void setUsers(ArrayList<VKUser> users) {
        this.users = users;
    }

    public String getPhoto50() {
        return photo50;
    }

    public void setPhoto50(String photo50) {
        this.photo50 = photo50;
    }

    public String getPhoto100() {
        return photo100;
    }

    public void setPhoto100(String photo100) {
        this.photo100 = photo100;
    }

    public String getPhoto200() {
        return photo200;
    }

    public void setPhoto200(String photo200) {
        this.photo200 = photo200;
    }

    public VKConversation.State getState() {
        return state;
    }

    public void setState(VKConversation.State state) {
        this.state = state;
    }

    public VKConversation.Type getType() {
        return type;
    }

    public void setType(VKConversation.Type type) {
        this.type = type;
    }
}

File: app/src/main/java/ru/melodin/fast/api/model/VKChat.kt
package ru.melodin.fast.api.model

import org.json.JSONObject
import ru.melodin.fast.util.ArrayUtil
import java.io.Serializable
import java.util.*

class VKChat : VKModel, Serializable {

    var id: Int = 0
    var title: String? = null
    var adminId: Int = 0
    var users = ArrayList<VKUser>()
    var photo50: String? = null
    var photo100: String? = null
    var photo200: String? = null

    var state: VKConversation.State = VKConversation.State.IN
    var type: VKConversation.Type? = VKConversation.Type.USER

    constructor()

    constructor(o: JSONObject) {
        this.id = o.optInt("id", -1)
        this.title = o.optString("title")
        this.adminId = o.optInt("admin_id", -1)

        val users = o.optJSONArray("users")
        if (!ArrayUtil.isEmpty(users))
            this.users = VKUser.parse(users!!)

        this.photo50 = o.optString("photo_50")
        this.photo100 = o.optString("photo_100")
        this.photo200 = o.optString("photo_200")
        this.state = if (o.has("left")) VKConversation.State.LEFT else if (o.has("kicked")) VKConversation.State.KICKED else VKConversation.State.IN
        this.type = VKConversation.getType(o.optString("type"))
    }

    companion object {
        private const val serialVersionUID = 1L

        fun getIntState(state: VKConversation.State?): Int {
            if (state == null) return -1
            return when (state) {
                VKConversation.State.IN -> 0
                VKConversation.State.LEFT -> 1
                VKConversation.State.KICKED -> 2
            }
        }

        fun getState(state: Int): VKConversation.State {
            return when (state) {
                0 -> VKConversation.State.IN
                1 -> VKConversation.State.LEFT
                2 -> VKConversation.State.KICKED
                else -> VKConversation.State.IN
            }
        }
    }
}
